core/src/main/java/com/graphhopper/util/GHUtility.java

```diff
@@ -465,7 +465,7 @@ public static EdgeIteratorState getEdge(Graph graph, int base, int adj) {
     public static int createEdgeKey(int nodeA, int nodeB, int edgeId, boolean reverse) {
         edgeId = edgeId << 1;
         if (reverse)
-            return (nodeA > nodeB) ? edgeId : edgeId + 1;
+            return (nodeA >= nodeB) ? edgeId : edgeId + 1;
```

Comment:
Ah, this is a nice fix! (is it not related to this issue?)

Reply:
Yes this is one of the two fixes needed here. Imagine a loop edge with `nodeA==nodeB`. If the edge key is not independent of `reverse` the bwd search does not find the `SPTEntry` of the fwd search in `updateBestPath` and vice versa.